scripts/release/asset_publish.py

```diff
@@ -277,33 +286,66 @@ def _str2bool(v: str) -> bool:
             )
 
             # Handle specific asset types
-            if asset.type in [assets.AssetType.COMPONENT, assets.AssetType.ENVIRONMENT]:
-                # Assemble command
-                cmd = assemble_command(
-                    asset.type.value, str(asset.spec_with_path),
-                    registry_name, final_version, resource_group, workspace, debug_mode)
-                # Run command
-                run_command(cmd, failure_list, debug_mode)
+            if asset.type == assets.AssetType.COMPONENT:
+                # load component and check if environment exists
+                logger.print(f"spec's path: {asset.spec_with_path}")
+                component = load_component(asset.spec_with_path)
+                env = component.environment
+                ws_env_pattern = re.compile("^(.+):(.+)$")
+                registry_env_pattern = re.compile("^azureml://registries/.+/environments/(.+)/versions/(.+)")
+                match = registry_env_pattern.search(env)
+                if not match:
+                    match = ws_env_pattern.search(env)
```

Comment:
Nit: You probably want `match()` instead.
```suggestion
                match = registry_env_pattern.match(env)
                if not match:
                    match = ws_env_pattern.match(env)
```

Reply:
Thanks for suggestion. Done